spec.emu

```diff
@@ -74,6 +74,36 @@ location: https://github.com/tc39/proposal-measure/
     <emu-clause id="sec-amount-abstract-ops-decimal-digit-strings">
       <h1>Operations on Decimal Digit Strings</h1>
 
+      <emu-clause id="sec-amount-canonicalizedigitstring" type="abstract operation">
+        <h1>CanonicalizeDigitString(
+          _s_: a decimal digit String
+        ): a decimal digit String
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It ensures that a decimal digit string adheres to a restricted syntax in which there are no redundant leading zeroes, no leading or trailing decimal points, and ensures that any *"E"* (indicating exponential notation) is downcased.</dd>
+        </dl>
+        <emu-alg>
+          1. If _s_ contains an occurrence of *"E"*, then
+            1. Let _beforeE_ be the substring of _s_ before the first occurrence of *"E"*.
+            1. Let _afterE_ be the substring of _s_ after the first occurrence of *"E"*.
+            1. Set _s_ to the string concatenation of _beforeE, *"e"*, and _afterE_.
+          1. If _s_ contains an occurrence of *"e"*, then
+            1. Let _beforeE_ be the substring of _s_ before the first occurrence of *"e"*.
+            1. Let _afterE_ be the substring of _s_ after the first occurrence of *"E"*.
+            1. Let _canonicalizedBeforeE_ be CanonicalizeDigitString(_beforeE_).
+            1. Let _canonicalizedAfterE_ be CanonicalizeDigitString(_afterE_).
+            1. Return the string concatenation of _canonicalizedBeforeE_, *"e"*, and _canonicalizedAfterE_.
+          1. If _s_ begins with an occurrence of *"00"*, then
+            1. Let _afterDoubleZero_ be the substring of _s_ after the first occurrence of *"00"*.
+            1. Let _oneFewerZero_ be the string concatenation of *"0"* and _afterDoubleZero_.
+            1. Return CanonicalizeDigitString(_oneFewerZero_).
```

Comment:
So the canonical representation of "0001" is "001", and not "1"? Why are any leading zeros left in?

Oh, wait, the function is called multiple times? This seems silly.